Source/Core/Core/GeckoCode.cpp

```diff
@@ -153,71 +160,95 @@ static Installation InstallCodeHandlerLocked()
     }
   }
 
-  u32 codelist_base_address;
-  u32 codelist_end_address;
+  u32 codelist_base_address = INSTALLER_BASE_ADDRESS + (u32)data.length() - 8;
+  u32 codelist_end_address = INSTALLER_END_ADDRESS;
+
+  // Write a magic value to 'gameid' (codehandleronly does not actually read this).
+  PowerPC::HostWrite_U32(0xd01f1bad, INSTALLER_BASE_ADDRESS);
+
   // Let the Gecko codehandler use free space from Melee's tournament mode region
-  if (SConfig::GetInstance().GetGameID() == "GALE01")
+  if (SConfig::GetInstance().m_melee_version == Melee::NTSC ||
+      SConfig::GetInstance().m_melee_version == Melee::MEX)
   {
-    INFO_LOG(ACTIONREPLAY,
-      "Detected GALE01 - using tournament mode region for Gecko codelist");
+    // Here we are replacing a line in the codehandler with a blr.
+    // The reason for this is that this is the section of the codehandler
+    // that attempts to read/write commands for the USB Gecko. These calls
+    // were sometimes interfering with the Slippi EXI calls and causing
+    // the game to loop infinitely in EXISync.
+    PowerPC::HostWrite_U32(0x4E800020, 0x80001D6C);
+
+    // Write GCT loader into memory which will eventually load the real GCT into the heap
+    std::string bootloaderData;
+    std::string _bootloaderFilename = File::GetSysDirectory() + GCT_BOOTLOADER;
+    if (!File::ReadFileToString(_bootloaderFilename, bootloaderData))
+    {
+      OSD::AddMessage("bootloader.gct not found in Sys folder.", 30000, 0xFFFF0000);
+      ERROR_LOG(ACTIONREPLAY, "Could not enable cheats because bootloader.gct was missing.");
+      init_failed = true;
+      return Installation::Failed;
+    }
 
-    // Set codelist base to the tournament mode region
-    codelist_base_address = 0x801910E0;
-    codelist_end_address = 0x8019AF4C;
+    if (bootloaderData.length() > codelist_end_address - codelist_base_address)
+    {
+      OSD::AddMessage("Gecko bootloader too large.", 30000, 0xFFFF0000);
+      ERROR_LOG(SLIPPI, "Gecko bootloader too large");
+      init_failed = true;
+      return Installation::Failed;
+    }
 
-    // Patch codehandler to use tournament mode region
-    PowerPC::HostWrite_U32(0x3DE08019, 0x80001904); // lis r15, 0x8019
-    PowerPC::HostWrite_U32(0x61EF10E0, 0x80001908); // ori r15, r15, 0x10e0
+    // Install bootloader gct
+    for (size_t i = 0, e = bootloaderData.length(); i < e; ++i)
```

Comment:
pretty sure you can do range based loop for std::string

Reply:
i don't think that is better here because we need `i` to compute the address we want to write to.

I do think `e = bootloaderData.length()` is unnecessary.

so how about `for (size_t i = 0; i < bootloaderData.length(); ++i)`?

Reply:
that's fine, or make an additional variable is how it's usually done

```
index = 0;
for (const auto &x: some_stuff) {
    // stuff using index and x
    index++
}
```

kind of gratuitious for this specific case but just context

as long as `e` is gone im good with this whole pr

Reply:
this was a port of a PR I wrote. Honestly I don't remember why I added e here. Either it was a copy pasta or I was using it in the loop originally. <shrug>